libcaf_net/caf/net/basp/application.hpp

```diff
@@ -73,7 +74,8 @@ class application {
       return err;
     auto hdr = to_bytes(header{message_type::handshake,
                                static_cast<uint32_t>(buf_.size()), version});
-    parent.write_packet(hdr, buf_);
+    std::vector<byte> header(hdr.begin(), hdr.end());
```

Comment:
Since `to_bytes` returns a `std::array<byte, header_size>` I had to copy the content.
Is there some deeper thought in using std::array other than the fixed size of `header` or can I refactor that function too?